### PR TITLE
feat(e2e): parallel-daemon isolation, dual-backend matrix, virtio debug, stress runner

### PR DESCRIPTION
**Pull request by AprilNEA**
Stacked on #362 (uses its `DaemonHandle`/signing fixtures — merge that first; this PR's own diff is the last 6 commits).

Completes the harness-first assessment's question 2 (parallel fix independence) and question 3 (peripheral engineering, by ROI):

## Q2 — parallel daemons
- `DaemonHandle::spawn` now refuses a data dir inside `~/.arcbox` / `~/.arcbox-dev`, so a daemon under test can never contend with a real daemon for the flock/sockets/machine state.
- The convention (one worktree/agent = one data dir) and the three host-global resources it does not cover (privileged helper, `~/.docker` context, published host ports) are documented in `tests/e2e/README.md`.

## Q3.1 — dual-backend matrix (VZ as oracle)
- `VmDefaults.backend` (`vz`/`hv`, default `vz`): first-boot System VM backend selectable non-interactively via `ARCBOX_VM_BACKEND` or `config.toml`; the persisted machine backend still wins after a switch.
- New `backend_matrix` e2e test runs the boot-assets Docker lifecycle once per backend; the verdict distinguishes HV-only failures (HV implementation) from double failures (layers above the hypervisor).

## Q3.2 — virtio queue observability
- `VirtioMmioState` counts per-queue guest kicks and raised interrupts (survive device reset).
- `DeviceManager::virtio_debug()` → `Vmm::virtio_debug()`: one structured record per device with live avail/used ring indices and EVENT_IDX slots read from guest memory (empty under VZ).
- Exposed as `SystemService.GetVirtioDebug`; the harness dumps `virtio-debug.json` automatically when a scenario fails, while the VM is still alive. The `hv_e2e` probe prints the same snapshot on phase failure.

## Q3.3 — stress runner
- `cargo xtask e2e --backend vz|hv|both --repeat N [--test …] [--fail-fast]`: prebuilds once, runs the ignored e2e test repeatedly, archives per-run logs under `target/e2e-artifacts/<unix-time>/`, and records each failing run's preserved data dir.

## Q3.5 — performance numbers
- `RunMetrics`: per-run phase timings (daemon ready, image pull, lifecycle tests; hv_e2e create/start/agent-ready/stop) written as JSON to the run's data dir and to the runner's artifacts archive.

## Verification
- `cargo fmt --check`, `cargo clippy --workspace --exclude arcbox-agent --all-targets -- -D warnings`: clean.
- `cargo test --workspace --exclude arcbox-agent`: 1615 passed, 0 failed (24 new tests: isolation guard, backend config parsing, kick/interrupt counters, ring snapshot, metrics).
- Not run here: live VM runs of `backend_matrix`/`hv_vmm` and a real `GetVirtioDebug` against a booted HV VM — no boot assets/HV environment on this machine.

### CODE REVIEW

**Comment by AprilNEA** [comment]
Live validation on this Apple Silicon machine (macOS 26.4, dev boot assets 0.6.1, Developer ID signing):

| Check | Result |
|---|---|
| `daemon_failure` | ✅ PASS — FAILED phase surfaced in ~2 s with the daemon's own 404 error |
| `hv_vmm` (bare-VMM HV probe) | ✅ PASS — boot, vsock RPC, DAX, supervision, pause/resume, stop |
| `virtio_debug` (new, this PR) | 🔴 boot fails, forensics work — see below |
| `backend_matrix` via `cargo xtask e2e` | runner + verdict logic verified; both runs red for environmental reasons below |
| VZ daemon boot | ✅ READY in 14.7 s; Docker API reachable end-to-end (pull then fails: guest-egress TLS handshake timeout to registry-1.docker.io — host reaches it fine; OSS-daemon-without-helper networking gap) |
| HV daemon boot | 🔴 deterministic 4/4: readiness `early eof`. The captured `virtio-debug.json` shows all three VirtioFs devices at **zero kicks** (guest never sent FUSE_INIT → agent on /arcbox never ran), balanced blk rings, and 3121 vsock RSTs — first structured evidence of the known OSS-daemon virtiofs gap; the same machine passes `hv_vmm` including its DAX phase, so the gap is daemon-context-specific |

Two fixes in this PR came out of the live run:

- `GetVirtioDebug` was gated on the general runtime-ready check, i.e. rejected exactly during a stuck boot — its main use case. The daemon now publishes a separate `early_runtime` handle right after runtime construction, and the machine-level snapshot no longer requires `Running`.
- An isolated test data dir could never satisfy the daemon's agent fallback (the CDN ships no standalone agent); `stage_dev_boot_assets` now stages the freshest locally available `arcbox-agent` into `boot/<version>/`.

**Comment by AprilNEA** [comment]
**HV root-fix campaign update — the cold-boot bug is found, fixed, and the HV daemon now boots and runs Docker end to end.**

Root cause (ABX-386, full analysis on the Linear issue): `VirtioMmioState`'s queue register file was fixed at 8 slots while virtio-blk configures one queue per vCPU. On >8-vCPU VMs, guest blk-mq queues 8..N never existed host-side; boot-critical I/O submitted from CPU ≥ 8 vanished, leaving PID 1 in a demand-paging D-state. Commit `06b1c92a` sizes the register file to `MAX_VIRTQUEUES = 64` (+ spec-correct `QueueNumMax = 0` beyond it, regression test).

Second find while validating: the HV guest had no wall clock (no RTC device; boot never pinged, so the agent's clock-sync never ran) — every TLS handshake failed with `current time 2020-01-01`. Commit `76c08184` pushes host time via one ping after readiness.

Validation on this machine (18-core Apple Silicon):
- bare probe: 13/13 boots pass incl. the full daemon-shape config (pre-fix: 12/16/18-vCPU configs fail; ≤8 pass)
- **HV daemon → READY: 10/10** (pre-fix 0/4)
- **HV full Docker lifecycle (pull via ECR, run, logs, exec, stop, rm): 3/3** — first time ever
- HV `daemon_ready` 11.1 s vs VZ baseline 14.7 s — HV now beats VZ to READY

Also in this push: config-matrix knobs on `hv_e2e` (the bisect tool), per-vCPU exit counters + broadcast-kick accounting in `GetVirtioDebug` (R2/R3's measurement; first snapshot already shows 2301 unpark-broadcasts in one boot), `cargo xtask idle` sampler, and `ARCBOX_E2E_IMAGE` override.

Baselines recorded for the remaining perf routes (all measured with the new tooling): VM-level agent-ready 2.7–9.2 s (target <1.5 s → R3/R5), idle CPU 3.87% (target <0.05% → R3), idle RSS 1.04 GB (target <150 MB → balloon). Known separate items: daemon exits with SIGSEGV on SIGTERM teardown after a successful run (stop-path family, ABX-364/367); VZ-backend pulls fail on this network through Apple's NAT while the HV datapath's own NAT works.

File: virt/arcbox-vmm/src/device/debug.rs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        clippy::cast_ptr_alignment,
+        reason = "pointer alignment is checked at runtime just above the cast"
+    )]
+    fn read_guest_u16(&self, gpa: u64) -> Option<u16> {
+        let base = self.guest_ram_base?;
+        if gpa == 0 {
+            return None;
+        }
+        let offset = gpa.checked_sub(self.guest_ram_gpa)?;
+        if offset + 2 > self.guest_ram_size as u64 {
```
[the remark this answers is not in the document]
Comment (AprilNEA) [reply virt/arcbox-vmm/src/device/debug.rs]: Confirmed real: `offset + 2` could wrap in release (bypassing the bounds check into the unsafe read) and the callers' `avail + 2` / `avail + 4 + 2*size` sums could overflow-panic in debug — on exactly the garbage-ring input this snapshot exists for. Fixed in 1669c80f: the field-offset math moved inside `read_guest_u16(base_gpa, field_offset)` and every step (`checked_add`, `checked_sub`, `checked_add(2)`) now returns `None` on overflow. Regression test `virtio_debug_survives_garbage_ring_addresses` programs near-`u64::MAX` ring addresses and asserts the snapshot survives with unreadable fields.

[the remark this answers is not in the document]
Comment (AprilNEA) [reply virt/arcbox-vmm/src/device/debug.rs]: Real — same finding as the thread above. Fixed in 1669c80f with fully checked GPA arithmetic inside `read_guest_u16` plus a near-`u64::MAX` regression test.

File: tests/e2e/src/daemon.rs [hunk 2]
```diff
@@ -239,6 +262,38 @@ impl Drop for DaemonHandle {
...
+    let resolved = data_dir.canonicalize().unwrap_or_else(|_| {
+        data_dir
+            .parent()
+            .and_then(|p| p.canonicalize().ok())
+            .and_then(|p| data_dir.file_name().map(|n| p.join(n)))
+            .unwrap_or_else(|| data_dir.to_path_buf())
+    });
+    for profile in [ArcboxProfile::Production, ArcboxProfile::Development] {
+        let root = profile.default_data_dir();
+        if resolved.starts_with(&root) {
```
[the remark this answers is not in the document]
Comment (AprilNEA) [reply tests/e2e/src/daemon.rs]: Real edge case: a symlinked `~/.arcbox` root compared only literally would let a data dir addressed via the symlink target through. Fixed in 1669c80f — the guard now also compares against each profile root's canonicalized form (`starts_with` on both literal and canonical).

File: app/arcbox-core/src/vm_lifecycle/boot.rs [hunk 3]
```diff
@@ -153,6 +154,30 @@ impl LifecycleShared {
...
+    /// the guest clock stays at the kernel default epoch and every TLS
+    /// handshake fails certificate validity checks. Best effort: a failed
+    /// sync only warns (readiness already proved the agent reachable, and
+    /// the guest can also be synced by any later ping).
+    async fn sync_guest_clock(&self) {
+        let machine_manager = self.machine_manager.clone();
+        let machine_name = self.machine_name.clone();
+        let result = tokio::task::spawn_blocking(move || {
+            let mut agent = machine_manager.connect_agent(&machine_name)?;
+            agent.ping_blocking().map(|_| ())
```
[the remark this answers is not in the document]
Comment (AprilNEA) [reply app/arcbox-core/src/vm_lifecycle/boot.rs]: Confirmed: `connect_agent` yields an async transport on AF_VSOCK (VZ) and Linux, so `ping_blocking` failed deterministically there and every VZ boot logged a spurious warning. Fixed in 1669c80f: `sync_guest_clock` branches on `agent.is_blocking()` — blocking ping on HV, async `ping().await` on VZ/Linux — so the clock push now also works over async transports instead of only warning.
